Add Common Cure Rights Commitment

richardfontana opened the pull request:
Following an earlier discussion some time ago with @bmbouter I am proposing adoption of the Common Cure Rights Commitment by the Pulp project.

richardfontana committed:
Add Common Cure Rights Commitment

bmbouter commented:
@richardfontana Thank you for this great writeup. I want to check my understanding.

1. This does not modify the license, so it's not a re-licensing.
2. It is a commitment to the core rights language included in the PUP. This commitment lives above and beyond the existing GPLv2 license?

richardfontana commented:
@bmbouter I do not see it as a relicensing, since it affects one procedural aspect of license enforcement. Rather, I see it as the grant of an additional permission or exception to users beyond what is granted in GPLv2 itself. It is somewhat like a GPLv2 project deciding to adopt, say, an OpenSSL linking exception.

I am not sure if I understand your second question , but the idea is that that commitment would be placed in a project repository in the same directory as the project copy of GPLv2 -- for example, in the top-level directory at github.com/pulp/pulp.

bmbouter commented on pup-0005.md:
```diff
@@ -0,0 +1,148 @@
+
+## Displaying the CCRC
+
+A file named COMMITMENT (or some alternative name) will be added to
```
Can the "or some alternative name" be removed? To be adopted we need the name to be named here in the PUP.

ipanova replied:
+1

bmbouter replied:
I pushed a commit with this revision to the PR so this is fixed. See that specific change here: https://github.com/richardfontana/pups/commit/99fcd35e1cc396a1ba5a34555f093304dd07a333

bmbouter commented:
I proposed a comment period and eventually a vote on this via the mailing list. See the post [here](https://www.redhat.com/archives/pulp-dev/2018-May/msg00027.html).

Pulp has plugins and "core", so I think it needs a clarification on the scope of this decision. Plugins are beyond our control both in terms of license and CCRC. One way to say it may be that it's scope to apply to "GPLv2 licensed software assets for Pulp's core". Alternative wording/ideas are welcome. You can see this aspect mentioned as the last FAQ line to the list too.

@richardfontana If there is anything I can do to help or work through these questions, please let me know.

ipanova commented on pup-0005.md:
```diff
@@ -0,0 +1,148 @@
+    from that copyright holder, and you cure the violation prior to 30
+    days after your receipt of the notice.
+
+We intend this Commitment to be irrevocable, and binding and
```
don't we want to specify some time here? proposer can not be bound indefinitely and any kind of revocation is without the effect in case there will be the need of change/modification
irrevocable right without any time limitation is a supreme and a very strong promise afaik

bmbouter replied:
I missed this comment in my review. I'm confused at what specifying time here would intend to do. Also who is the proposer here? I overall don't understand this comment.

ipanova replied:
proposer is legal term, usually it;s a party which formally suggests a plan, law, legislation etc. [0]
In this specific case - we are proposer.

w/r to 'irrevocable right/agreement/contract/etc' ( also a legal term) by default it is something that cannot be altered, withdrawn or terminated and if no date or period is specified of its' validity then you are bound to this indefinitely.
[0] https://www.ldoceonline.com/dictionary/proposer

i thought maybe @richardfontana could answer this question, maybe i am wring since i do not remember all the clauses.

RCMariko replied:
@ipanova - I read that as our intentions are that "we mean it. we're not going to randomly back out of our commitment and give you no opportunity to cure the violations."

Commit:
Clarify display language

The CRCC recommended displaying the COMMITMENT file in the root of the
repo, but also allowed for "an alternative" To make the proposal
concrete enough to adopt I removed the "or some alternative name"
language so it's specific.

I also changed the tense of the recommendation to other Pulp
repositories to present test instead of future. That way the merged PUP
will be that recommendation instead of a followon action.

richardfontana commented:
@bmbouter @ipanova sorry for the late response.  The point about plugins normally being outside scope makes sense. 
@ipanova the irrevocability language is intentional. The wording is the same in the corporate commitment and (if I remember correctly) was adapted from the revised version of the Red Hat Patent Promise. @RCMariko is correct as to the interpretation. Note that what's irrevocable is the commitment as to some particular code that has been released and licensed. It would be possible to cease making the commitment in the future as to code that doesn't exist now but is created in the future. So in that sense there is a limit. But otherwise, it is intentionally designed to be something that is binding and which recipients of the code can rely upon.

bmbouter commented:
With no blocking votes, one +0, and 5 +1 votes this PUP has passed.
